core/primitives/src/transaction.rs

```diff
@@ -1,1029 +1,492 @@
-    /// Account ID of the account who signed the initial transaction.
-    pub originator_id: AccountId,
-    /// The public key used to sign the initial transaction.
+#[derive(Hash, Serialize, Deserialize, PartialEq, Eq, Clone, Debug)]
+pub struct DeleteKeyAction {
     pub public_key: PublicKey,
 }
 
-impl Callback {
-    pub fn new(
-        method_name: Vec<u8>,
-        args: Vec<u8>,
-        amount: Balance,
-        refund_account: AccountId,
-        originator_id: AccountId,
-        public_key: PublicKey,
-    ) -> Self {
-        Callback {
-            method_name,
-            args,
-            results: vec![],
-            amount,
-            callback: None,
-            result_counter: 0,
-            refund_account,
-            originator_id,
-            public_key,
-        }
-    }
-}
-
-impl TryFrom<receipt_proto::Callback> for Callback {
+impl TryFrom<transaction_proto::Action_DeleteKey> for DeleteKeyAction {
     type Error = Box<dyn std::error::Error>;
 
-    fn try_from(proto: receipt_proto::Callback) -> Result<Self, Self::Error> {
-        Ok(Callback {
-            method_name: proto.method_name,
-            args: proto.args,
-            results: proto
-                .results
-                .into_iter()
-                .map(|value| if value.len() > 0 { Some(value) } else { None })
-                .collect::<Vec<_>>(),
-            amount: proto.amount.unwrap_or_default().try_into()?,
-            callback: proto.callback.into_option().map(|value| value.into()),
-            result_counter: proto.result_counter as usize,
-            refund_account: proto.refund_account,
-            originator_id: proto.originator_id,
-            public_key: PublicKey::try_from(&proto.public_key as &[u8])?,
-        })
+    fn try_from(a: transaction_proto::Action_DeleteKey) -> Result<Self, Self::Error> {
+        Ok(DeleteKeyAction { public_key: proto_to_type(a.public_key)? })
     }
 }
 
-impl From<Callback> for receipt_proto::Callback {
-    fn from(callback: Callback) -> Self {
-        receipt_proto::Callback {
-            method_name: callback.method_name,
-            args: callback.args,
-            results: RepeatedField::from_iter(
-                callback.results.iter().map(|value| value.clone().unwrap_or(vec![])),
-            ),
-            amount: SingularPtrField::some(callback.amount.into()),
-            callback: SingularPtrField::from_option(callback.callback.map(|value| value.into())),
-            result_counter: callback.result_counter as u32,
-            refund_account: callback.refund_account,
-            originator_id: callback.originator_id,
-            public_key: callback.public_key.as_ref().to_vec(),
+impl From<DeleteKeyAction> for transaction_proto::Action_DeleteKey {
+    fn from(a: DeleteKeyAction) -> transaction_proto::Action_DeleteKey {
+        transaction_proto::Action_DeleteKey {
+            public_key: SingularPtrField::some(a.public_key.into()),
+
```

Comment:
I liked it here, because it separates proto fields from default field. And I don't want to use default to not accidentally skip some new fields later.

Reply:
Added newlines in other places for consistency.